LNK-21-Leenk: 생일 축하 UI 구현 + API 연결

nabbang6 opened the pull request:
## 📝 Work Description

### 컴포넌트

- `ProfileImageWithFallback.tsx` - 생일자 여부 확인 후, 생일 모자 표시되게 리팩토링

### 부가 기능 페이지 (private.tsx)

- 생일자 카드 / 곧 생일자 카드 / 생일 편지 카드 컴포넌트 구현
- 생일 편지 전송 시 표시되는 `ReportModal.tsx` → `TextInputModal.tsx`로 리팩토링
- 생일 편지 전송 완료 시 표시되는 `ImageModal.tsx` 구현
- 받은 편지 조회 페이지(`/birthday/letters`) 구현
- 생일자 조회, 곧 생일자 조회, 받은 편지 조회, 생일 편지 전송, 편지 읽음 처리 api 구현

### 마이페이지 (mypage.tsx/edit.tsx)

- `ProfileCard.tsx`에 생일 조회 기능 추가
- `CalendarButton.tsx`에 생일 모드 추가하여 리팩토링
- 마이페이지 정보 수정 시 생일 수정 기능 추가

### 온보딩 (profile.tsx)

- 생일 입력 단계 추가


---

### 수정사항 
- 부가 기능 관련을 private -> extra로 네이밍 변경

- MBTI/생일 정보가 false일 때 '????' -> '미등록'으로 표시되게 변경
<img width="300" alt="image" src="https://github.com/user-attachments/assets/f3ea8dd7-d67d-4ec1-b89a-9efbdfb2fd1c" />

- 생일 편지 전송 완료 모달에서 사용자 이름 뒤 2글자만 표시
<img width="300" alt="image" src="https://github.com/user-attachments/assets/ad69d63e-8c46-4646-8fbb-a6d984a785dc" />


<br />

## 📸 Screenshot

- 생일 편지 전송 
https://github.com/user-attachments/assets/87bd0e60-8f35-46eb-80e9-32397d346e3e


- 생일자/곧 생일자가 없는 경우 엠티 페이지
<img width="300" alt="image" src="https://github.com/user-attachments/assets/f34483ef-4f9c-4063-9a8f-f847688de152" />


- 받은 편지 조회 페이지
<p>
<img width="300" alt="image" src="https://github.com/user-attachments/assets/e06f41b9-2a8b-4883-a927-e891d9cef174" />
<img width="300" alt="image" src="https://github.com/user-attachments/assets/6f4d6e92-bdde-4c02-a6b1-9c76543a98ff" />
</p>


- 마이페이지 내 생일 조회/입력
https://github.com/user-attachments/assets/debc02e4-c300-4e49-8431-f73f3acd7246

<br />

## 🚨Issue

목데이터로는 전부 테스트 완료하긴 했는데... 
제 테스트 환경으로는 회원가입 진행이 안 되는 상황이라 (현재 안드로이드 상으로는 안 되는,,게 맞는 거겟죠?..) 생일 편지 조회(`/birthday/letters/me`), 편지 읽음 처리(`/birthday/letters/me/mark) api가 실제로 잘 작동되는지 추후 확인이 필요합니다,,

나머지 api는 잘 작동되는 거 확인했습니다! 

<br />

## 📣 To Reviewers

코드&파일 변경 수가 생각보다 너무....많아졌더라구요.... ㅠ___ㅠ
다음부턴 꼭 이슈 더 나눠서 pr 올려보겠습니다.........

그리고 현재 받은 편지 조회 페이지 경로 `/birthday/letters`로 설정해두었는데, 추후 부가 기능이 늘어날 것을 생각하면`/private/birthday/letters`가 나을지... 조금 고민 중인데 역시 후자가 나을까요??

<br />

## ✅ Check List
- [x] 코드가 정상적으로 컴파일되나요?
- [x] merge할 브랜치의 위치를 확인했나요?
- [x] Label을 지정했나요?

<br />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **새로운 기능**
  * 생일 정보 등록·편집 및 프로필에 생일 저장 지원
  * 오늘 생일 배지 표시 및 사용자별 생일 여부 노출
  * 생일 축하 편지 작성·전송, 수신 편지함 조회 및 읽음 표시(마크) 지원
  * 오늘/예정된 생일자 목록과 전용 페이지(목록·카드·상세) 추가
  * 캘린더 기반 생일 선택 UI, 관련 모달(입력/이미지/완료) 및 아이콘 자산 추가
  * 마이페이지·피드·작성자 UI에 생일 관련 표시 및 편의성 개선
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

nabbang6 committed:
design: 부가 기능 svg 교체

nabbang6 committed:
feat: 프로필 이미지 상단에 생일 모자 표시 기능 추가

nabbang6 committed:
feat: 생일 여부(isUserBirthdayToday) 표시 속성 추가

nabbang6 committed:
feat: 생일자 카드 컴포넌트 추가

nabbang6 committed:
feat: 곧 생일자 카드 컴포넌트 추가

nabbang6 committed:
feat: 생일 편지 컴포넌트 추가

nabbang6 committed:
refactor: 마이페이지 내 생일 기능 추가

nabbang6 committed:
refactor: 온보딩 내 생일 기능 추가

nabbang6 committed:
feat: birthday 타입 정의 및 zustand 스토어 생성

nabbang6 committed:
feat: 생일 편지 완료용 모달(ImageModal) 컴포넌트 생성

nabbang6 committed:
refactor: ReportModal -> TextInputModal로 변경, 생일 편지 작성 모드 추가

nabbang6 committed:
feat: 부가 기능 페이지에 생일 기능 추가

nabbang6 committed:
feat: 받은 편지 조회 페이지 구현

nabbang6 committed:
refactor: birthday API를 get/post 단위로 통합

nabbang6 committed:
refactor: 편지 읽음 처리 위치를 private -> letters 페이지로 이동

nabbang6 committed:
fix: 생일 카드 클릭 시 선택한 사용자 이름이 ImageModal에 반영되게 수정

nabbang6 committed:
style: margin/padding 값 width 기준으로 변경

nabbang6 committed:
style: ImageModal 애니메이션을 fade로 변경

dalzzy commented on app/(page)/private.tsx:
```diff
@@ -1,49 +1,174 @@
+const Title = styled.Text`
+  color: ${colors.text[1]};
+  font-family: ${fonts.ExtraBold};
+  font-size: ${fontSize['lg']}px;
```
전체적으로 `fontSize.lg` 이렇게 적용하신 것도 있고, `fontSize['lg']` 이렇게 적용하신 것도 있던데 특별한 이유가 있는 건지 궁금합니다!

nabbang6 replied:
어라랏 그러게요... `fontSize.lg` < 요 형식으로 통일되게 수정했습니다!

dalzzy commented on components/mypage/ProfileCard.tsx:
```diff
@@ -48,9 +52,24 @@ export default function ProfileCard({
+
+            <InfoText textColor={colors.text[3]}>생일</InfoText>
+            <InfoText textColor={colors.primary} isPrimary>
+              {birthday ? formatMonthDay(birthday) : '????'}
```
별 건 아니지만 제 생각에는,, 값이 false 일 때 ????로 표시되는 것보다 "정보 없음" 같은 텍스트가 나을 것 같은데 어떻게 생각하시나용?? 
????로 표시되면 정보가 없는건지 오류인건지 헷갈릴 수도 있을 것 같아서요..!!

nabbang6 replied:
아항 그렇네요.. 좋은 것 같습니다 !! bb 
'정보 없음' 보다 '미등록' 처리해두는 게 쪼금 더 직관적일 것 같아서 (정보 자체가 없다기보다는 사용자가 정보를 미등록햇다는 의미가 강조되게..) 한번 요걸로 반영해볼게요!

<img width="404" height="448" alt="image" src="https://github.com/user-attachments/assets/b6e38673-4839-4599-be86-1150707b94ce" />

dalzzy commented on components/private/BirthdayCard.tsx:
```diff
@@ -0,0 +1,136 @@
+import ProfileImageWithFallback from '../feed/ProfileImageWithFallback';
+import CustomButton from '../common/Button/CustomButton';
```
이것도 절대경로에서 import하도록 수정하면 깔끔할 것 같아요!!

dalzzy commented on types/birthday.ts:
```diff
@@ -0,0 +1,35 @@
+  userId: number;
+  thumbnail: string;
+  name: string;
+  isUserBirthdayToday: boolean;
+}
+
+export interface BirthdayData {
+  users: BirthdayUser[];
+  myBirthdayLettersCounts: number;
+  hasNewLetters: boolean;
+}
+
+export interface BirthdayLetter {
+  letterId: number;
+  author: BirthdayUser;
+  message: string;
+  createdAt: string;
+}
+
+export interface UpcomingBirthdayUser {
+  userId: number;
+  thumbnail: string;
+  name: string;
+  isUserBirthdayToday: boolean;
+  birthday: string;
```
BirthdayUser와 UpcomingBirthdayUser가 거의 동일한 구조를 갖고 있어
UpcomingBirthdayUser가 BirthdayUser를 상속받도록 하면 
중복 제거와 타입 재활용 측면에서 조금 더 깔끔할 것 같습니다!

이런 식으로용!!
```
export interface UpcomingBirthdayUser extends BirthdayUser {
  birthday: string;
}
```

dalzzy commented on app/(page)/extra.tsx:
이게 아마 원래 잠겨있는 페이지라 private으로 해둔 것 같은데, 이제 부가 기능 페이지니까 다른 이름으로 바꾸는 게 어떨까용?? 
으음 extra 나 features 이런 걸로 하면 괜찮을 것 같애요@@! 아님 생각나는 좋은 네이밍이 있다면,, 그것도 좋구욤!

nabbang6 replied:
앗 좋습니당!! extra가 좀 더 직관적인 것 같아서 좋은 것 같아요 👍,, 
경로랑 폴더명 쪽도 같이 쭉 반영해놓을게요~!

dalzzy reviewed (approved):
꺅 수고하셨습니다~!! 폴더 구조도 그렇고,, 코드도 좀 더러운 부분이 있어서 쉽지 않으셨을 것 같은데 너무 잘 구현해주신 것 같습니다 👍🏻👍🏻  코드래빗이 남긴 것도 한번 확인해주시면 좋을 것 같아용

받은 편지 조회 페이지 경로는 저도 한 depth 더 들어가는 게 좋은 것 같습니다!

nabbang6 committed:
style: fontSize 접근 방식 통일

nabbang6 committed:
fix: 모달이 닫힐 때 onClose, closeModal 둘 다 호출되도록 변경

nabbang6 committed:
chore: 절대경로에서 import되도록 수정

nabbang6 committed:
refactor: UpcomingBirthdayUser가 BirthdayUser를 상속하도록 변경

nabbang6 committed:
refactor: 부가 기능 관련 폴더/파일명 private -> extra로 변경

nabbang6 committed:
chore: console.log 삭제

nabbang6 committed:
refactor: MBTI/생일 정보가 false일 때 '????' -> '미등록'으로 표시되게 변경

nabbang6 committed:
fix: 부가 기능 페이지(extra) 개선
- 페이지 포커스 시 userInfo 새로 가져오도록 refetch 추가
- ImageModal에서 closeModal 호출을 모달 내부로 이동 (extra 페이지에서는 상태 초기화만 처리)

nabbang6 committed:
refactor: 생일 편지 전송 완료 모달에서 사용자 이름 뒤 2글자만 표시

nabbang6 committed:
Merge branch 'develop' of https://github.com/Leets-Makers/Leenk-MOBILE into feat/LNK-21-Leenk-생일-축하-UI-구현

nabbang6 committed:
refactor: 부가 기능 페이지(extra) 개선
- handleBirthdayCardPress 의존성 배열 수정
- BirthdayCard에 조건부로 데이터 전달되게 수정
- 이름 길이 안전성 검사
- gap 속성 제거
